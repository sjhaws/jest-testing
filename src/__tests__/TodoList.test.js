import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../TodoList';
import toJson from 'enzyme-to-json';

describe(<TodoList />, () => {

  describe('render', () => {
    let component;
    beforeEach( () => {
      component = shallow(<TodoList />);
    });

    it ('matches snapshot', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  describe('functionality', () => {
    let component;
    beforeEach( () => {
      component = shallow(<TodoList />);
    });

    it('updates state on change', () => {
      let input = component.find('input')
      input.simulate('focus')
      input.simulate('change', {target: {name: 'name', value: 'Hello'}})
      expect(component.state('name')).toEqual('Hello')
    })
  })
});