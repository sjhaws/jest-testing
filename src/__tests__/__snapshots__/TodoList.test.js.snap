// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] render matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TodoList)>
    <TodoList
      dispatch={[Function]}
      items={Array []}
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Add Item"
            required={true}
            value=""
          />
        </form>
        <hr />
        <ul />
      </div>
    </TodoList>
  </Connect(TodoList)>
</Provider>
`;
